### PR TITLE
refactor:(make) dont download binaries if theyre already present

### PR DESCRIPTION
**Pull request by shreddedbacon**
# General Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [ ] PR title is ready for inclusion in changelog

# Database Migrations

- [ ] If your PR contains a database migation, it **MUST** be the latest in date order alphabetically

#3811 fixed some issues symbolic linking, this incorporates some additional checks to prevent needlessly downloading the same versions of these binaries over and over again if they are already present.

Also includes some commands to clean up old `lagoon-chart` directories and k3dconfig temp files.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -414,10 +414,12 @@ ifeq ($(KUBECTL_VERSION), $(shell kubectl version --short --client 2>/dev/null |
 	$(info linking local kubectl version $(KUBECTL_VERSION))
 	ln -sf $(shell command -v kubectl) ./local-dev/kubectl
 else
+ifeq ($(KUBECTL_VERSION), $(shell kubectl version --short --client 2>/dev/null | sed -E 's/Client Version: v([0-9.]+).*/\1/'))
```
Comment (tobybellwood) [inline Makefile]: Should this not be neq the local-dev version?

Comment (shreddedbacon) [reply Makefile]: :see_no_evil:

Comment (shreddedbacon) [reply Makefile]: Turns out there is more than just that wrong with the `local-dev/kubectl` check

**Review by tobybellwood — APPROVED** [review]
Works nicely!

**Pull request merged**